spatial extension fails to compile with strict options

trying to build experimental package i've seen this: 

    8%] Building CXX object src/CMakeFiles/sbml-static.dir/sbml/packages/spatial/extension/SpatialCompartmentPlugin.cpp.o
    In file included from /io/python-libsbml/libsbml_source/src/sbml/packages/spatial/common/CompressionUtil.cpp:47:0:
    /io/python-libsbml/libsbml_source/src/sbml/packages/spatial/common/CompressionUtil.h: In function ‘type* readSamplesFromString(const string&, size_t&)’:
    /io/python-libsbml/libsbml_source/src/sbml/packages/spatial/common/CompressionUtil.h:91:53: error: there are no arguments to ‘malloc’ that depend on a template parameter, so a declaration of ‘malloc’ must be available [-fpermissive]
         type* data = (type*)malloc(sizeof(type) * length);
                                                         ^
    /io/python-libsbml/libsbml_source/src/sbml/packages/spatial/common/CompressionUtil.h:91:53: note: (if you use ‘-fpermissive’, G++ will accept your code, but allowing the use of an undeclared name is deprecated)
    /io/python-libsbml/libsbml_source/src/sbml/packages/spatial/common/CompressionUtil.cpp: In function ‘void compress_data(void*, size_t, int, unsigned char*&, int&)’:
    /io/python-libsbml/libsbml_source/src/sbml/packages/spatial/common/CompressionUtil.cpp:166:58: error: ‘malloc’ was not declared in this scope
       result = (unsigned char*)malloc(sizeof(int) * outLength);
                                                              ^
    /io/python-libsbml/libsbml_source/src/sbml/packages/spatial/common/CompressionUtil.cpp: In function ‘void uncompress_data(void*, size_t, double*&, size_t&)’:
    /io/python-libsbml/libsbml_source/src/sbml/packages/spatial/common/CompressionUtil.cpp:236:54: error: ‘malloc’ was not declared in this scope
       result = (double*)malloc(sizeof(double) * outLength);
                                                          ^
    /io/python-libsbml/libsbml_source/src/sbml/packages/spatial/common/CompressionUtil.cpp: In function ‘void uncompress_data(void*, size_t, int*&, size_t&)’:
    /io/python-libsbml/libsbml_source/src/sbml/packages/spatial/common/CompressionUtil.cpp:306:48: error: ‘malloc’ was not declared in this scope
       result = (int*)malloc(sizeof(int) * outLength);
                                                    ^
    /io/python-libsbml/libsbml_source/src/sbml/packages/spatial/common/CompressionUtil.cpp: In function ‘void copySampleArrays(double*&, size_t&, double*, size_t)’:
    /io/python-libsbml/libsbml_source/src/sbml/packages/spatial/common/CompressionUtil.cpp:319:57: error: ‘malloc’ was not declared in this scope
       target = (double*)malloc(sizeof(double) * sourceLength);
                                                             ^
    /io/python-libsbml/libsbml_source/src/sbml/packages/spatial/common/CompressionUtil.cpp: In function ‘void copySampleArrays(int*&, size_t&, int*, size_t)’:
    /io/python-libsbml/libsbml_source/src/sbml/packages/spatial/common/CompressionUtil.cpp:328:51: error: ‘malloc’ was not declared in this scope
       target = (int*)malloc(sizeof(int) * sourceLength);
                                                       ^
    /io/python-libsbml/libsbml_source/src/sbml/packages/spatial/common/CompressionUtil.cpp: In function ‘void copySampleArrays(int*&, size_t&, unsigned char*, size_t)’:
    /io/python-libsbml/libsbml_source/src/sbml/packages/spatial/common/CompressionUtil.cpp:337:51: error: ‘malloc’ was not declared in this scope
       target = (int*)malloc(sizeof(int) * sourceLength);
                                                       ^

--- a/src/sbml/packages/spatial/common/CompressionUtil.h
+++ b/src/sbml/packages/spatial/common/CompressionUtil.h
@@ -41,6 +41,7 @@
 
 #include <vector>
 #include <iostream>
+#include <cstdlib>
 
 LIBSBML_CPP_NAMESPACE_BEGIN
 

